app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import {useEffect, useState} from "react";
+
+export default function Greeting() {
+  const [message, setMessage] = useState("");
+  const [flag, setFlag] =useState(false);
+  useEffect(() => {
+    async function getData()  {
+      const res = await fetch("http://localhost:3000/messages/random");
+      const data = await res.json();
+      setMessage(data.content)
+    }
+    getData()
+  }, [flag]);
```

Comment:
- According to the project requirements, you are to implement and integrate redux for managing your app's state, I suggest you make use of the [redux-toolkit](https://redux-toolkit.js.org/). Please implement the following and ensure that your API calls and state management are all done in redux:
  - [The createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) function which helps you to make API calls to your backend.
  - [The createSlice](https://redux-toolkit.js.org/api/createSlice) function that automatically generates action creators and action types that correspond to the reducers and state.
  - Redux store to help manage your app's state.